trinity/protocol/eth/servers.py

```diff
@@ -0,0 +1,167 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+    Set,
+    Type,
+    cast,
+)
+
+from cancel_token import CancelToken
+
+from eth.exceptions import (
+    HeaderNotFound,
+)
+from eth_typing import (
+    Hash32,
+)
+
+from p2p import protocol
+from p2p.peer import BasePeer
+from p2p.protocol import (
+    Command,
+)
+
+from trinity.db.chain import AsyncChainDB
+from trinity.protocol.common.servers import BaseRequestServer, BasePeerRequestHandler
+from trinity.protocol.eth import commands
+from trinity.protocol.eth.peer import ETHPeer, ETHPeerPool
+
+from eth.rlp.receipts import Receipt
+from eth.rlp.transactions import BaseTransactionFields
+from eth.tools.logging import TraceLogger
+
+from trinity.protocol.eth.constants import (
+    MAX_BODIES_FETCH,
+    MAX_RECEIPTS_FETCH,
+    MAX_STATE_FETCH,
+)
+from trinity.protocol.eth.requests import HeaderRequest as ETHHeaderRequest
+from trinity.rlp.block_body import BlockBody
+
+
+class ETHPeerRequestHandler(BasePeerRequestHandler):
+    def __init__(self, db: AsyncChainDB, logger: TraceLogger, token: CancelToken) -> None:
```

Comment:
This is more inherited code, but I suspect the idea was that this handler isn't "big enough" to justify its own logger (like no one would want to tweak the log level of only this handler). Also, since it's not a service, it doesn't get one for free. Just speculating though. I don't have any strong feelings on how it should be.

Reply:
I'm generally :-1: on this approach.  I have in a semi-anti-pattern bucket simply because it makes testing harder.